fdbserver/TLogGroup.actor.h

```diff
@@ -78,13 +82,24 @@ class TLogGroupCollection : public ReferenceCounted<TLogGroupCollection> {
 	// and group size set in the parent class.
 	void recruitEverything();
 
+	// Find a TLogGroup for assigning a storage team.
+	TLogGroupRef selectFreeGroup();
+
+	// Assigns a storage team to given group
+	bool assignStorageTeam(ptxn::StorageTeamID teamId, TLogGroupRef group);
+
 	// Add mutations to store state to given txnStoreState transaction request 'recoveryCommitReq'.
 	void storeState(CommitTransactionRequest* recoveryCommitReq);
 
 	// Loads TLogGroupCollection state from given IKeyValueStore, which will be txnStoreState passed
 	// by master.
 	void loadState(const Standalone<RangeResultRef>& store);
 
+	// Start monitoring storage teams by reading \xff keyspace.
```

Comment:
It seems that currently the master is the role monitors team changes. Looking at the implementation of this function, it is assigning new team to a tlog group. I think there are two problems here:

1. It's monitoring team changes every 1s on master. We don't want to do a lot of work on master, if the work can be performed by other roles.
2. The current logic is: team changes are written to `\xff` key space; master monitors changes; master issue transactions to assign the team to a group; all proxies pick up the assignment.

Instead of doing this, I propose the following logic:

1. Team changes are written to `\xff` key space;
2. The Proxy got this transaction processes it as metadata transaction, and perform the team to tlog group assignment;
3. All other proxies pick up the change with metadata transaction mechanism via Resolvers.

This avoids problem 1, so master doesn't need to do anything, and seems more efficient than the current logic.

BTW, the tlog group information should be propagated from master to all commit proxies before they accept commits. Commit proxies need to map from a key to team and then to the tlog group.